app/controllers/application_controller.rb

```diff
@@ -2,4 +2,13 @@ class ApplicationController < ActionController::Base
   # Prevent CSRF attacks by raising an exception.
   # For APIs, you may want to use :null_session instead.
   protect_from_forgery with: :exception
+
+private
+
+  def error_unprocessable_entity
+    render \
+      file: Rails.root.join('public', '422.html'),
+      layout: false,
+      status: 422
```

Comment:
My first thought was the same, that this might be better as an exception, but it wasn't obvious how to do it. Your comment has prompted me to look in more detail, and I think I've found a way to do it.